### ISSUE TITLE
Switch to C++20 compiler

### ISSUE THREAD

**Issue by KeyWorksRW** [KeyWorksRW/wxUiEditor#723]
### Description:

Currently the code requires a C++17 compiler. As of this writing, Linux packages default to gcc version 9 and clang version 10, both of which support C++17. However, that means clang is 4 versions behind, and gcc is 3 versions behind, so neither of them support C++20. I've avoided upgrading the code base to C++20 because the standard linux distributions are so far behind.

The code base makes extensive use of `ttLib::cstr` which is a class derived from `std::string`. The `cstr` class provides methods such as `contains()` and `is_sameprefix()` both of which have equivalents in C++20. Using the ttLib derivative classes makes perfect sense in our private repositories -- but in a public repository like this one, it lessens the usability of the code. By that, I mean that a reader can't just copy and paste a snippet of the code to another project because they would also have to pick up the ttLib code as well.

If we upgrade to C++20, we should be able to reduce the use of the ttLib derivative classes, using the `std` versions instead. There are also several other additions to C++20 that would increase the readability of the code base, and in some cases we might be able to reduce memory footprint due to the new `constains()` working on all associative containers.

The main downside is that building on Linux is going to require the user to get a newer tools package rather than the standard package. Ironically, I can't build on Linux right now anyway because gcc 9 doesn't like the ctor in ttLib::cstr. If I upgraded the code to C++20, we might not need ttLib::cstr, but then I can't build anyway because both gcc and clang are too old in the standard tools packages.

The C++20 Linux problems will go away 3-5 years from now. Switching now means fewer people can clone the project and build it in the short term, but it makes the code more readable and more maintainable by someone not familiar with the ttLib classes.

**Comment by Randalphwa** [comment]
Actually, `contains()` for `std::string` and `std::string_view` didn't make it into C++20, but is instead in C++23. Also, `ttLib::cstr` is the best way to get a class that has many of the same methods as `wxString` but uses utf8 on Windows instead of utf16, making it considerably more efficient. In addition, our version of `contains` allows for a case-insensitive comparison, which is not the case for the C++23 version.

Bottom line: using std::string_view instead of ttLib::cview is preferable. ttLib::sview is the next best choice for passing around string pointers. `ttLib::cstr` is arguably still better than `std::string` since it acts a lot like wxString (such as appending use `<<` operator).

**Comment by KeyWorksRW** [comment]
This is essentially done, at least as far as a full-blown refactor pass.
